src/data/move.ts

```diff
@@ -7722,6 +7752,64 @@ export function applyMoveChargeAttrs(attrType: Constructor<MoveAttr>, user: Poke
   return applyMoveChargeAttrsInternal((attr: MoveAttr) => attr instanceof attrType, user, target, move, args);
 }
 
+/**
+ * Base class defining all {@linkcode selectableConditions}
+ * Is used to add {@linkcode UserMoveConditionFunc} in order to check if move can be selected
+ */
+export class MoveSelectCondition {
+  protected func: UserMoveConditionFunc;
+
+  constructor(func: UserMoveConditionFunc) {
+    this.func = func;
+  }
+
+  /**
+   * Checks condition and adds appropriate MoveRestrictionTag accordingly
+   * @param user {@linkcode Pokemon} that uses the move
+   * @param move {@linkcode Move} that is used
+   * @returns true if Tag was added successfully
+   */
+  apply(user: Pokemon, move: Move): boolean {
+    return true;
+  }
+}
+
+/**
+ * extends {@linkcode MoveSelectCondition} and contains the condition
+ * for {@link https://bulbapedia.bulbagarden.net/wiki/Stuff_Cheeks_(move) | Stuff Cheeks}s success
+ */
+export class StuffCheeksCondition extends MoveSelectCondition {
+  constructor() {
+    super((user, move) => this.selectableCondition(user));
+  }
+
+  /**
+   * Checks if the user is holding a berry
+   * @param user {@linkcode Pokemon} whose berries to check
+   * @returns true if the user is holding a berry, otherwise false
+   */
+  private selectableCondition(user: Pokemon): boolean {
+    return user.getHeldItems().filter(m => m instanceof BerryModifier, user.isPlayer()).length > 0;
```

Comment:
```suggestion
    return user.getHeldItems().filter(m => m instanceof BerryModifier).length > 0;
```
The original function this is copied from was `scene.findModifiers()` which took a `isPlayer: boolean` param, but `.filter()` is a different function and passing `user.isPlayer()` to it doesn't do anything here.

Reply:
Thanks for the hint!